core/rpc/src/system/mod.rs

```diff
@@ -115,4 +119,15 @@ impl<B: traits::Block> SystemApi for System<B> {
 			Ok(health)
 		}
 	}
+
+	fn system_peers(&self) -> Result<Vec<PeerInfo>> {
+		Ok(self.sync.peers().into_iter().map(|(idx, peer_id, p)| PeerInfo {
+			index: idx as usize,
+			peer_id: peer_id.map_or("".into(), |p| p.to_base58()),
+			roles: format!("{:?}", p.roles),
+			protocol_version: p.protocol_version,
+			best_hash: format!("{:?}", p.best_hash),
+			best_number: format!("{}", p.best_number),
```

Comment:
We do have `B` as a block generic parameter here, I'd prefer to return the exact types instead of their `Debug/Display` versions (i.e. make `PeerInfo` generic over block or these two types).